x-pack/plugins/secops/server/lib/kpi_network/elasticsearch_adapter.ts

```diff
@@ -20,16 +26,55 @@ export class ElasticsearchKpiNetworkAdapter implements KpiNetworkAdapter {
     request: FrameworkRequest,
     options: RequestBasicOptions
   ): Promise<KpiNetworkData> {
+    const generalQuery: KpiNetworkESMSearchBody[] = buildGeneralQuery(options);
+    const uniqueSourcePrivateIpsQuery: KpiNetworkESMSearchBody[] = buildUniquePrvateIpQuery(
+      'source',
+      options
+    );
+    const uniqueDestinationPrivateIpsQuery: KpiNetworkESMSearchBody[] = buildUniquePrvateIpQuery(
+      'destination',
+      options
+    );
     const response = await this.framework.callWithRequest<KpiNetworkHit, TermAggregation>(
       request,
-      'search',
-      buildQuery(options)
+      'msearch',
+      {
+        body: [
+          ...generalQuery,
+          ...uniqueSourcePrivateIpsQuery,
+          ...uniqueDestinationPrivateIpsQuery,
+        ],
+      }
     );
 
     return {
-      networkEvents: getOr(null, 'hits.total.value', response),
-      uniqueFlowId: getOr(null, 'aggregations.unique_flow_id.value', response),
-      activeAgents: getOr(null, 'aggregations.active_agents.value', response),
+      networkEvents: getOr(null, 'responses.0.hits.total.value', response),
+      uniqueFlowId: getOr(null, 'responses.0.aggregations.unique_flow_id.value', response),
+      activeAgents: getOr(null, 'responses.0.aggregations.active_agents.value', response),
+      uniquePrivateIps: this.combineUniquePrivateIp(response),
     };
   }
+
+  private combineUniquePrivateIp(
+    response: DatabaseMultiResponse<KpiNetworkHit, TermAggregation>
+  ): number | null {
+    const uniqueSourcePrivateIp = getOr(
+      null,
+      'responses.1.aggregations.unique_private_ips.value',
+      response
+    );
+    const uniqueDestinationPrivateIp = getOr(
+      null,
+      'responses.2.aggregations.unique_private_ips.value',
+      response
+    );
+    if (!isNil(uniqueSourcePrivateIp) && !isNil(uniqueDestinationPrivateIp)) {
+      return uniqueSourcePrivateIp + uniqueDestinationPrivateIp;
```

Comment:
So this metric would be the sum of the unique sources and destinations?

Combining the metrics could make it look like you have ~2x as many private IPs on your network than what you really do. How about we show only the cardinality of the `source.ip`? I think that would give you a rough estimate of the clients on your network.

Or maybe show both numbers in the same box

```
+--------------------+
|   38 sources       |
|    4 destinations  |
|                    |
|   Private IPs      |
+--------------------+
```

Reply:
That's a good point @andrewkroh! I think it relevant to the look and feel to the product as well,  would be great to involve the team in this discussion. Shall we keep this for now and raise this question in the next team meeting?